pkg/gateway/handler/gateway.go

```diff
@@ -0,0 +1,20 @@
+package handler
+
+import (
+	"net/http"
+
+	coreHttp "github.com/skygeario/skygear-server/pkg/core/http"
+)
+
+func NewGatewayHandler() http.HandlerFunc {
```

Comment:
I think TenantAuthzMiddleware, GatewayHandler and DeploymentRouteHandler can be merged as one? The headers written by TenantAuthzMiddleware is intended to be used internally. If they are merged, no internal headers need to be written.

Reply:
See if c980027 looks good:)